src/mbgl/geometry/line_atlas.cpp

```diff
@@ -54,13 +54,17 @@ void addRoundDash(
     if (ranges.empty()) return;
 
     for (int y = -n; y <= n; y++) {
-        int row = yOffset + n + y;
+        int row = static_cast<int32_t>(yOffset) + n + y;
         const uint32_t index = image.size.width * row;
         uint32_t currIndex = 0;
         DashRange range = ranges[currIndex];
 
         for (uint32_t x = 0; x < image.size.width; ++x) {
-            if (x / range.right > 1.0f && ++currIndex < ranges.size()) {
+            if (range.right == 0) {
```

Comment:
nit: if (range.right == 0 && x != 0 && ++currIndex < ranges.size())

Reply:
The nested condition can not be combined as we need to check if `range.right == 0` or not, if combined, for case `range.right == 0 && x == 0`, if will enter another if check, the runtime error can not be avoided.